src/DI/OrmExtension.php

```diff
@@ -129,26 +130,42 @@ public function loadEntityManagerConfiguration(): void
 			throw new InvalidStateException(sprintf('EntityManagerDecorator class "%s" not found', $entityManagerDecoratorClass));
 		}
 
-		// Entity Manager
-		$original = $builder->addDefinition($this->prefix('entityManager'))
-			->setType(DoctrineEntityManager::class)
-			->setFactory(DoctrineEntityManager::class . '::create', [
-				$builder->getDefinitionByType(Connection::class), // Nettrine/DBAL
-				$this->prefix('@configuration'),
-			])
-			->setAutowired(false);
-
-		// Entity Manager Decorator
-		$builder->addDefinition($this->prefix('entityManagerDecorator'))
-			->setFactory($entityManagerDecoratorClass, [$original]);
-
-		// ManagerRegistry
-		$builder->addDefinition($this->prefix('managerRegistry'))
-			->setType(ManagerRegistry::class)
-			->setArguments([
-				$builder->getDefinitionByType(Connection::class),
-				$this->prefix('@entityManagerDecorator'),
-			]);
+		foreach ($builder->findByType(\Doctrine\DBAL\Connection::class) as $definitionName => $serviceDefinition) {
+			$match = Strings::match($definitionName, '~([a-zA-Z]+\.([a-zA-Z]+))\.connection~');
+
+			if ($serviceDefinition->getTag(\Nettrine\DBAL\DI\DbalExtension::TAG_CONNECTION) !== null && array_key_exists(1, $match) && array_key_exists(2, $match)) {
+				$nameWithPrefix = $match[1];
+				$name = $match[2];
+
+			} else {
+				continue;
+			}
+
+			// Entity Manager
+			$original = $builder->addDefinition($this->prefix($name . '.entityManager'))
+				->setType(DoctrineEntityManager::class)
+				->setFactory(DoctrineEntityManager::class . '::create', [
+					$builder->getDefinition($nameWithPrefix . '.connection'), // Nettrine/DBAL
+					$this->prefix('@configuration'),
+				])
+				->setAutowired(false);
+
+			$autowired = $name === \Nettrine\DBAL\DI\DbalExtension::DEFAULT_CONNECTION_NAME ? true : false;
+
+			// Entity Manager Decorator
+			$builder->addDefinition($this->prefix($name . '.entityManagerDecorator'))
```

Comment:
EntityManagerDecorator depends only on EntityManager, not on specific connection. It should be outside of foreach.